libvirt_manager/extra_net_config - nmstate format

hjensas opened the pull request:
Add a new parameter to libvirt_manager role:
  cifmw_libvirt_manager_extra_network_configuration

When set the raw nmstate configuration defined in the parameter is added using the `ci_nmstate` after creating the libvirt bridges. One usecase is to add VLAN interfaces on the hypervisor.

For example:
```
cifmw_libvirt_manager_extra_network_configuration:
  interfaces:
    - name: "vlan{{ cifmw_networking_definition.networks.internalapi.vlan }}"
      type: vlan
      state: up
      vlan:
        base-iface: cifmw-osp_trunk
        id: "{{ cifmw_networking_definition.networks.internalapi.vlan }}"
        protocol: 802.1q
      ipv4:
        enabled: true
        dhcp: false
        address:
          - ip: "{{ cifmw_networking_definition.networks.internalapi.gateway }}"
            prefix-length: "{{ cifmw_networking_definition.networks.internalapi.network | ansible.utils.ipaddr('prefix') }}"
    - name: "vlan{{ cifmw_networking_definition.networks.internalapi1.vlan }}"
      type: vlan
      state: up
      vlan:
        base-iface: cifmw-trunk1
        id: "{{ cifmw_networking_definition.networks.internalapi1.vlan }}"
        protocol: 802.1q
      ipv4:
        enabled: true
        dhcp: false
        address:
          - ip: "{{ cifmw_networking_definition.networks.internalapi1.gateway }}"
            prefix-length: "{{ cifmw_networking_definition.networks.internalapi1.network | ansible.utils.ipaddr('prefix') }}"
```

pablintino commented:
/approve

lewisdenny reviewed (changes requested):
@hjensas 

Could you please add `cifmw_libvirt_manager_extra_network_configuration` to one of the existing molecule jobs to add testing for this feature?

hjensas commented:
> @hjensas
> 
> Could you please add `cifmw_libvirt_manager_extra_network_configuration` to one of the existing molecule jobs to add testing for this feature?

I added test to the `ocp_layout` scenario.

```
TASK [List all interfaces] *****************************************************
Monday 14 October 2024  08:07:08 -0400 (0:00:00.068)       0:01:05.537 ********
changed: [instance]

TASK [Ensure the VLAN interfaces was created] **********************************
Monday 14 October 2024  08:07:08 -0400 (0:00:00.214)       0:01:05.751 ********
ok: [instance] => (item=vlan20)
ok: [instance] => (item=vlan21)
```

lewisdenny commented:
> > @hjensas
> > 
> > Could you please add `cifmw_libvirt_manager_extra_network_configuration` to one of the existing molecule jobs to add testing for this feature?
> 
> I added test to the `ocp_layout` scenario.
> 
> ```
> TASK [List all interfaces] *****************************************************
> Monday 14 October 2024  08:07:08 -0400 (0:00:00.068)       0:01:05.537 ********
> changed: [instance]
> 
> TASK [Ensure the VLAN interfaces was created] **********************************
> Monday 14 October 2024  08:07:08 -0400 (0:00:00.214)       0:01:05.751 ********
> ok: [instance] => (item=vlan20)
> ok: [instance] => (item=vlan21)
> ```

Legend,  thanks mate!

/lgtm
